In [1]:
import pandas as pd
df = pd.read_csv("../data/ips_malveillantes.csv")

df.shape
df["countryCode"].value_counts().head(10)
df["abuseConfidenceScore"].describe()

count    1000.0
mean      100.0
std         0.0
min       100.0
25%       100.0
50%       100.0
75%       100.0
max       100.0
Name: abuseConfidenceScore, dtype: float64

## Observations sur le dataset AbuseIPDB

- 1000 adresses IP recuperees (limite du plan gratuit par requete).
- Toutes ont un score de confiance de 100 (abuseConfidenceScore), 
  min = max = 100, ecart-type = 0.
- Le parametre confidenceMinimum=90 utilise dans le script est en realite 
  redondant : l'API AbuseIPDB trie par score decroissant, donc les 1000 
  premieres entrees de la blacklist sont deja toutes a confiance maximale.
- Cette blacklist represente donc uniquement les IPs les "pires" 
  (confiance absolue), pas un echantillon representatif de toutes 
  les IPs suspectes.

In [2]:
df["countryCode"].value_counts().head(10)

countryCode
US    211
CN     83
NL     77
FR     62
DE     61
IN     39
SG     39
HK     35
GB     35
CA     31
Name: count, dtype: int64

## Répartition géographique

Les 10 pays les plus représentés parmi les 1000 IPs malveillantes :
US (211), CN (83), NL (77), FR (62), DE (61), IN (39), SG (39), HK (35), 
GB (35), CA (31).

Cette concentration sur les US, la Chine et les Pays-Bas est cohérente 
avec la répartition mondiale des grands hébergeurs cloud, souvent 
exploités pour héberger des infrastructures malveillantes (VPS bon 
marché, absence de vérification d'identité stricte).